core/src/main/java/org/mule/runtime/core/internal/lock/InstanceLockGroup.java

```diff
@@ -151,8 +128,55 @@ public boolean hasPendingLocks() {
 
   @Override
   public void dispose() {
+    waitForLocksToBeUnlocked();
+  }
```

Comment:
We are no longer clearing up the Map (not even after the timeout expires), is this intentional?

Reply:
Yes, clearing the map would generate the same bug after the graceful shutdown timeout